compiler/docgen.nim

```diff
@@ -211,7 +211,8 @@ proc newDocumentor*(filename: AbsoluteFile; cache: IdentCache; conf: ConfigRef,
       writeFile(outp, importStmt & content)
       let c = if cmd.startsWith("nim "): os.getAppFilename() & cmd.substr(3)
               else: cmd
-      let c2 = c % quoteShell(outp)
+      let subst = if "-r" in c: "-r" else: "$1"
+      let c2 = c.replace(subst, "--verbosity:0 --hints:off " & subst) % quoteShell(outp)
```

Comment:
Suspicous code. Replace is not ``replacef``. Why not ``let c2 = (c & "--verbosity:0 --hints:off")% quoteShell(outp)"``?

Reply:
Can't put flags after the filename.

Reply:
Ok, but ``let subst = if "-r" in c: "-r" else: "$1"`` is suspicious. Why does ``-r`` lack the ``$1`` part?

Reply:
Not all tests are run so -r isn't common. But when it is, these flags have to go before it, hence looking for both.

Reply:
-r seems fragile, eg would trigger for `--rangeChecks:on`
a more robust solution would identify the position right before the filename in `nim cmd -flag1 -flag2 filename arg`, ideally by reusing same code compiler uses instead of duplicating logic

Reply:
Open to ideas but like I said, these params have to go before -r. How about `" -r "`?

Reply:
> these params have to go before -r

anything between nim binary and filename can go anywhere:
```
nim c --rangeChecks:on -r main
nim c  -r --rangeChecks:on main
nim --rangeChecks:on -r c main
```
> How about " -r "

not robust to things like:
```
nim c -r -d:mystr:"foo -r bar" main
```
(there are more such cases, eg --passC, --passL etc)

sure, these may be rare/edge cases, but I think it's just better to reuse (not re-implement) the logic nim compiler itself uses

Reply:
how about instead adding `--verbosity:0 --hints:off` to here:
```
tools/kochdocs.nim:235:27:    commands[i] = nim & " rst2html $# --git.url:$# -o:$# --index:on $#" %
```